From 8b12f2fcc9fb99cf8d6391449dd18404540fca7a Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Wed, 5 Nov 2014 09:38:43 +0800
Subject: [PATCH 538/635] MA-5837 - [sabresd_6dq]: support usb headset

The device for usb headset is AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET.
It's driver name is "USB-Audio", nedd bind the device with the driver.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 alsa/config_cs42888.h  | 3 +--
 alsa/config_usbaudio.h | 4 ++--
 alsa/config_wm8958.h   | 3 +--
 alsa/config_wm8962.h   | 3 +--
 alsa/tinyalsa_hal.c    | 7 +++++--
 5 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/alsa/config_cs42888.h b/alsa/config_cs42888.h
index c75fe20..162ccf1 100644
--- a/alsa/config_cs42888.h
+++ b/alsa/config_cs42888.h
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-/* Copyright (C) 2012 Freescale Semiconductor, Inc. */
+/* Copyright (C) 2012-2014 Freescale Semiconductor, Inc. */
 
 #ifndef ANDROID_INCLUDE_IMX_CONFIG_CS42888_H
 #define ANDROID_INCLUDE_IMX_CONFIG_CS42888_H
@@ -29,7 +29,6 @@ static struct audio_card  cs42888_card = {
             AUDIO_DEVICE_OUT_WIRED_HEADSET |
             AUDIO_DEVICE_OUT_WIRED_HEADPHONE |
             AUDIO_DEVICE_OUT_ANLG_DOCK_HEADSET |
-            AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET |
             AUDIO_DEVICE_OUT_ALL_SCO |
             AUDIO_DEVICE_OUT_DEFAULT ),
     .supported_in_devices = (
diff --git a/alsa/config_usbaudio.h b/alsa/config_usbaudio.h
index 8417b79..1c94061 100644
--- a/alsa/config_usbaudio.h
+++ b/alsa/config_usbaudio.h
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-/* Copyright (C) 2012 Freescale Semiconductor, Inc. */
+/* Copyright (C) 2012-2014 Freescale Semiconductor, Inc. */
 
 #ifndef ANDROID_INCLUDE_IMX_CONFIG_USBAUDIO_H
 #define ANDROID_INCLUDE_IMX_CONFIG_USBAUDIO_H
@@ -25,7 +25,7 @@
 static struct audio_card  usbaudio_card = {
     .name = "USB Device",
     .driver_name = "USB-Audio",
-    .supported_out_devices  = 0,
+    .supported_out_devices  = AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET,
     .supported_in_devices   = AUDIO_DEVICE_IN_USB_DEVICE,
     .defaults            = NULL,
     .bt_output           = NULL,
diff --git a/alsa/config_wm8958.h b/alsa/config_wm8958.h
index 1991515..6d9533f 100644
--- a/alsa/config_wm8958.h
+++ b/alsa/config_wm8958.h
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-/* Copyright (C) 2012 Freescale Semiconductor, Inc. */
+/* Copyright (C) 2012-2014 Freescale Semiconductor, Inc. */
 
 #ifndef ANDROID_INCLUDE_IMX_CONFIG_WM8958_H
 #define ANDROID_INCLUDE_IMX_CONFIG_WM8958_H
@@ -479,7 +479,6 @@ static struct audio_card  wm8958_card = {
             AUDIO_DEVICE_OUT_WIRED_HEADSET |
             AUDIO_DEVICE_OUT_WIRED_HEADPHONE |
             AUDIO_DEVICE_OUT_ANLG_DOCK_HEADSET |
-            AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET |
             AUDIO_DEVICE_OUT_ALL_SCO |
             AUDIO_DEVICE_OUT_DEFAULT),
     .supported_in_devices = (
diff --git a/alsa/config_wm8962.h b/alsa/config_wm8962.h
index dbb40a4..582be53 100644
--- a/alsa/config_wm8962.h
+++ b/alsa/config_wm8962.h
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-/* Copyright (C) 2012 Freescale Semiconductor, Inc. */
+/* Copyright (C) 2012-2014 Freescale Semiconductor, Inc. */
 
 #ifndef ANDROID_INCLUDE_IMX_CONFIG_WM8962_H
 #define ANDROID_INCLUDE_IMX_CONFIG_WM8962_H
@@ -206,7 +206,6 @@ static struct audio_card  wm8962_card = {
             AUDIO_DEVICE_OUT_WIRED_HEADSET |
             AUDIO_DEVICE_OUT_WIRED_HEADPHONE |
             AUDIO_DEVICE_OUT_ANLG_DOCK_HEADSET |
-            AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET |
             AUDIO_DEVICE_OUT_ALL_SCO |
             AUDIO_DEVICE_OUT_DEFAULT ),
     .supported_in_devices = (
diff --git a/alsa/tinyalsa_hal.c b/alsa/tinyalsa_hal.c
index 13050fd..f41cc69 100644
--- a/alsa/tinyalsa_hal.c
+++ b/alsa/tinyalsa_hal.c
@@ -522,8 +522,7 @@ static int start_output_stream_primary(struct imx_stream_out *out)
         select_output_device(adev);
     }
 
-    pcm_device = out->device & (AUDIO_DEVICE_OUT_ALL &
-                         ~(AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET | AUDIO_DEVICE_OUT_AUX_DIGITAL));
+    pcm_device = out->device & (AUDIO_DEVICE_OUT_ALL & ~AUDIO_DEVICE_OUT_AUX_DIGITAL);
     if (pcm_device && (adev->active_output[OUTPUT_ESAI] == NULL || adev->active_output[OUTPUT_ESAI]->standby)) {
         out->write_flags[PCM_NORMAL]            = PCM_OUT | PCM_MMAP;
         out->write_threshold[PCM_NORMAL]        = PLAYBACK_LONG_PERIOD_COUNT * LONG_PERIOD_SIZE;
@@ -949,6 +948,10 @@ static int out_set_parameters(struct audio_stream *stream, const char *kvpairs)
             if (out != adev->active_output[OUTPUT_HDMI]) {
                 adev->out_device = val;
                 out->device    = val;
+                if(out->device & AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET) {
+                    scan_available_device(adev, false);
+                }
+
                 select_output_device(adev);
             }
         }
-- 
1.8.0

